File: nomor 2/golang/ex9.go
package main

import "fmt"

var results [][]string

func arrange(array []string, memory []string) [][]string {
	var current string
	if memory == nil {
		memory = []string{}
	}
	for i := 0; i < len(array); i++ {
		current, array = array[i], append(array[:i], array[i+1:]...)
		if len(array) == 0 {
			results = append(results, append(memory, current))
		}
		arrange(append([]string(nil), array...), append(memory, current))
		array = append(array[:i], append([]string{current}, array[i:]...)...)
	}
	return results
}

func main() {
	candidates := []string{"Baswedan", "Subianto", "Maharani", "Ganjar"}
	chairs := arrange(candidates, nil)
	fmt.Println(chairs)
}

File: nomor 2/kotlin/ex9.kt
fun arrange(array: List<String>, memory: List<String>? = null): List<List<String>> {
    val results = mutableListOf<List<String>>()
    val tempMemory = memory ?: emptyList()
    for (i in array.indices) {
        val current = listOf(array[i])
        if (array.size == 1) {
            results.add(tempMemory + current)
        } else {
            val subarray = array.filterIndexed { index, _ -> index != i }
            val newMemory = tempMemory + current
            results.addAll(arrange(subarray, newMemory))
        }
    }
    return results
}

fun main() {
    val candidates = listOf("Baswedan", "Subianto", "Maharani","Ganjar")
    var chairs = arrange(candidates)
    for (chair in chairs) {
        println(chair)
    }
    
}
